### ISSUE TITLE
The result of 228 lines and 231 lines of train.py are same?

### ISSUE THREAD

**Issue by caijie12138** [EGiunchiglia/C-HMCNN#4]
train.py
The result of 228 line and 231 line are same?
Why not use 228 line result as the output directly？

**Comment by EGiunchiglia** [comment]
The results are not always the same. 
The operation done at line 231 tells the network to pass to the loss function:
1. the constrained output for those labels that have ground truth 0
2. the output of the network itself (not constrained) when the ground truth is 1 
This is what makes the difference between using the standard binary cross-entropy loss on top of the layer (which, as stated in the paper, might lead to wrong supervisions) and using our loss. 

Hope this was helpful :)
